File: backend/src/main/java/com/example/backend/model/VehicleFee.java
package com.example.backend.model;

import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "vehicle_fees")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class VehicleFee {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;

    @ManyToOne
    @JoinColumn(name = "vehicle_id", nullable = false)
    Vehicle vehicle;

    @Column(nullable = false, length = 7)
    String monthYear; // Format: 'YYYY-MM'

    @Column(nullable = false)
    Double amount;

    @Column(nullable = false)
    Boolean isPaid;

    LocalDate paidDate;

    String paidBy;

    String collectedBy;

    String createdBy;

    @CreationTimestamp
    LocalDateTime createdAt;
} 

File: frontend/src/types/vehicleFee.ts
export interface VehicleFee {
  id: number;
  vehicleId: number;
  monthYear: string;
  amount: number;
  isPaid: boolean;
  paidDate?: string;
  paidBy?: string;
  collectedBy?: string;
  createdBy: string;
  createdAt: string;
}

export interface VehicleFeeRequest {
  vehicleId: number;
  monthYear: string;
  amount: number;
}

export interface VehicleFeeResponse {
  success: boolean;
  message: string;
  data: VehicleFee;
}

export interface VehicleFeeListResponse {
  success: boolean;
  message: string;
  data: VehicleFee[];
} 